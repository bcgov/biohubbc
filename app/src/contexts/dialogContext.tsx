import IconButton from '@material-ui/core/IconButton';
import Snackbar from '@material-ui/core/Snackbar';
import CloseIcon from '@material-ui/icons/Close';
import { Color } from '@material-ui/lab/Alert';
import { ErrorDialog, IErrorDialogProps } from 'components/dialog/ErrorDialog';
import YesNoDialog, { IYesNoDialogProps } from 'components/dialog/YesNoDialog';
import React, { createContext, ReactNode, useState } from 'react';

export interface IDialogContext {
  /**
   * Set the yes no dialog props.
   *
   * Note: Any props that are not provided, will default to whatever value was previously set (or the default value)
   *
   * @memberof IDialogContext
   */
  setYesNoDialog: (props: Partial<IYesNoDialogProps>) => void;
  /**
   * The current yes no dialog props.
   *
   * @type {IYesNoDialogProps}
   * @memberof IDialogContext
   */
  yesNoDialogProps: IYesNoDialogProps;
  /**
   * Set the error dialog props.
   *
   * Note: Any props that are not provided, will default to whatever value was previously set (or the default value)
   *
   * @memberof IDialogContext
   */
  setErrorDialog: (props: Partial<IErrorDialogProps>) => void;
  /**
   * The current error dialog props.
   *
   * @type {IErrorDialogProps}
   * @memberof IDialogContext
   */
  errorDialogProps: IErrorDialogProps;
  /**
   * Set the snackbar props.
   *
   * Note: Any props that are not provided, will default to whatever value was previously set (or the default value)
   *
   * @memberof IDialogContext
   */
  setSnackbar: (props: Partial<ISnackbarProps>) => void;
  /**
   * The current snackbar props.
   *
   * @type {ISnackbarProps}
   * @memberof IDialogContext
   */
  snackbarProps: ISnackbarProps;
}

export interface ISnackbarProps {
  open: boolean;
  onClose: () => void;
  severity?: Color;
  color?: Color;
  snackbarMessage: ReactNode;
}

export const defaultYesNoDialogProps: IYesNoDialogProps = {
  dialogTitle: '',
  dialogText: '',
  open: false,
  onClose: () => {
    // default do nothing
  },
  onNo: () => {
    // default do nothing
  },
  onYes: () => {
    // default do nothing
  }
};

export const defaultErrorDialogProps: IErrorDialogProps = {
  dialogTitle: '',
  dialogText: '',
  open: false,
  onClose: () => {
    // default do nothing
  },
  onOk: () => {
    // default do nothing
  }
};

export const defaultSnackbarProps: ISnackbarProps = {
  snackbarMessage: '',
  open: false,
  onClose: () => {
    // default do nothing
  }
};

export const DialogContext = createContext<IDialogContext>({
  setYesNoDialog: () => {
    // default do nothing
  },
  yesNoDialogProps: defaultYesNoDialogProps,
  setErrorDialog: () => {
    // default do nothing
  },
  errorDialogProps: defaultErrorDialogProps,
  setSnackbar: () => {
    // default do nothing
  },
  snackbarProps: defaultSnackbarProps
});

/**
 * Wraps the provided children in a context that provides various modal dialogs/popups.
 *
 * @param {*} props
 * @return {*}
 */
export const DialogContextProvider: React.FC = (props) => {
  const [yesNoDialogProps, setYesNoDialogProps] = useState<IYesNoDialogProps>(defaultYesNoDialogProps);

  const [errorDialogProps, setErrorDialogProps] = useState<IErrorDialogProps>(defaultErrorDialogProps);

  const [snackbarProps, setSnackbarProps] = useState<ISnackbarProps>(defaultSnackbarProps);

  const setYesNoDialog = function (partialProps: Partial<IYesNoDialogProps>) {
    setYesNoDialogProps({ ...yesNoDialogProps, ...partialProps });
  };

  const setSnackbar = function (partialProps: Partial<ISnackbarProps>) {
    setSnackbarProps({ ...snackbarProps, ...partialProps });
  };

  const setErrorDialog = function (partialProps: Partial<IErrorDialogProps>) {
    setErrorDialogProps({ ...errorDialogProps, ...partialProps });
  };

  return (
    <DialogContext.Provider
      value={{
        setYesNoDialog,
        yesNoDialogProps,
        setErrorDialog,
        errorDialogProps,
        setSnackbar,
        snackbarProps
      }}>
      {props.children}
      <YesNoDialog {...yesNoDialogProps} />
      <ErrorDialog {...errorDialogProps} />
      <Snackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'center'
        }}
        open={snackbarProps.open}
        autoHideDuration={6000}
        onClose={() => setSnackbar({ open: false })}
        message={snackbarProps.snackbarMessage}
        action={
          <React.Fragment>
            <IconButton size="small" aria-label="close" color="inherit" onClick={() => setSnackbar({ open: false })}>
              <CloseIcon fontSize="small" />
            </IconButton>
          </React.Fragment>
        }
      />
    </DialogContext.Provider>
  );
};
