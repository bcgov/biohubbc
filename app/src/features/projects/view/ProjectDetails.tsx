import Box from '@material-ui/core/Box';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import FundingSource from 'features/projects/view/components/FundingSource';
import GeneralInformation from 'features/projects/view/components/GeneralInformation';
import IUCNClassification from 'features/projects/view/components/IUCNClassification';
import Partnerships from 'features/projects/view/components/Partnerships';
import ProjectCoordinator from 'features/projects/view/components/ProjectCoordinator';
import ProjectObjectives from 'features/projects/view/components/ProjectObjectives';
import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';
import { IGetProjectForViewResponse } from 'interfaces/useProjectApi.interface';
import React from 'react';
import ProjectPermits from './components/ProjectPermits';

export interface IProjectDetailsProps {
  projectForViewData: IGetProjectForViewResponse;
  codes: IGetAllCodeSetsResponse;
  refresh: () => void;
}

/**
 * Project details content for a project.
 *
 * @return {*}
 */
const ProjectDetails: React.FC<IProjectDetailsProps> = (props) => {
  const { projectForViewData, codes, refresh } = props;

  return (
    <>
      <Box component={Paper} p={3}>
        <Typography variant="h2">Project Details</Typography>
        <Box component="section" mt={1}>
          <GeneralInformation projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <ProjectObjectives projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <ProjectCoordinator projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <IUCNClassification projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <ProjectPermits projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <FundingSource projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
        <Box component="section" mt={3}>
          <Partnerships projectForViewData={projectForViewData} codes={codes} refresh={refresh} />
        </Box>
      </Box>
    </>
  );
};

export default ProjectDetails;
