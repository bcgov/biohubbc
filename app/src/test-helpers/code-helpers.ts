import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';

export const codes: IGetAllCodeSetsResponse = {
  coordinator_agency: [{ id: 1, name: 'A Rocha Canada' }],
  management_action_type: [{ id: 1, name: 'Management action' }],
  first_nations: [{ id: 1, name: 'First nations code' }],
  funding_source: [{ id: 1, name: 'Funding source code' }],
  investment_action_category: [{ id: 1, fs_id: 1, name: 'Investment action category' }],
  activity: [{ id: 1, name: 'Activity code' }],
  project_type: [{ id: 1, name: 'Project type' }],
  region: [{ id: 1, name: 'Region code' }],
  species: [{ id: 1, name: 'Species code' }],
  proprietor_type: [
    { id: 1, name: 'Proprietor code 1', is_first_nation: false },
    { id: 2, name: 'First Nations Land', is_first_nation: true }
  ],
  iucn_conservation_action_level_1_classification: [{ id: 1, name: 'IUCN class' }],
  iucn_conservation_action_level_2_subclassification: [{ id: 1, iucn1_id: 1, name: 'IUCN subclass 1' }],
  iucn_conservation_action_level_3_subclassification: [{ id: 1, iucn2_id: 1, name: 'IUCN subclass 2' }],
  system_roles: [
    { id: 1, name: 'Role 1' },
    { id: 2, name: 'Role 2' }
  ],
  project_roles: [
    { id: 1, name: 'Project Role 1' },
    { id: 2, name: 'Project Role 2' }
  ],
  administrative_activity_status_type: [
    { id: 1, name: 'Pending' },
    { id: 2, name: 'Actioned' },
    { id: 3, name: 'Rejected' }
  ],
  common_survey_methodologies: [
    { id: 1, name: 'Recruitment' },
    { id: 2, name: 'SRB' }
  ]
};
