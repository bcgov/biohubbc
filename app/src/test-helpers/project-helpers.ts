import { IGetProjectForViewResponse } from 'interfaces/useProjectApi.interface';

export const getProjectForViewResponse: IGetProjectForViewResponse = {
  id: 1,
  project: {
    project_name: 'Test Project Name',
    project_type: '1',
    start_date: '1998-10-10',
    end_date: '2021-02-26',
    project_activities: [1],
    completion_status: 'Active',
    publish_date: '2021-01-26'
  },
  permit: {
    permits: [
      {
        permit_number: '123',
        permit_type: 'Permit type'
      }
    ]
  },
  location: {
    location_description: 'Location description',
    geometry: []
  },
  objectives: {
    objectives: 'Et ad et in culpa si',
    caveats: 'sjwer bds'
  },
  coordinator: {
    first_name: 'Amanda',
    last_name: 'Christensen',
    email_address: 'amanda@christensen.com',
    coordinator_agency: 'Amanda and associates',
    share_contact_details: 'true'
  },
  iucn: {
    classificationDetails: [
      {
        classification: 'classification',
        subClassification1: 'sub classification 1',
        subClassification2: 'sub classification 2'
      },
      {
        classification: 'classification again',
        subClassification1: 'sub classification 1 again',
        subClassification2: 'sub classification 2 again'
      }
    ]
  },
  funding: {
    fundingSources: [
      {
        id: 0,
        agency_id: 1,
        agency_name: 'agency name',
        agency_project_id: 'ABC123',
        investment_action_category: 1,
        investment_action_category_name: 'investment action',
        funding_amount: 333,
        start_date: '2000-04-14',
        end_date: '2021-04-13',
        revision_count: 1
      }
    ]
  },
  partnerships: {
    indigenous_partnerships: ['partner 1', 'partner 2'],
    stakeholder_partnerships: ['partner 3', 'partner 4']
  }
};
