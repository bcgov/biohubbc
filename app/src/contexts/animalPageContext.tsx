import { useBiohubApi } from 'hooks/useBioHubApi';
import { useSurveyContext } from 'hooks/useContext';
import { useCritterbaseApi } from 'hooks/useCritterbaseApi';
import useDataLoader, { DataLoader } from 'hooks/useDataLoader';
import { ICritterDetailedResponse } from 'interfaces/useCritterApi.interface';
import { createContext, PropsWithChildren, useEffect, useMemo, useState } from 'react';

export interface ISurveyCritter {
  survey_critter_id: number;
  critterbase_critter_id: string;
}

/**
 * Context object that stores information about the survey Animal page.
 *
 * Note: this should only be used for data that is generally applicable to the entire page.
 *
 * @export
 * @interface IAnimalPageContext
 */
export type IAnimalPageContext = {
  isLoading: boolean;
  setIsLoading: (isLoading: boolean) => void;
  isDisabled: boolean;
  setIsDisabled: (isDisabled: boolean) => void;
  selectedAnimal: ISurveyCritter | null;
  setSelectedAnimal: (selectedAnimal: ISurveyCritter | null) => void;
  critterDataLoader: DataLoader<[critterbase_critter_id: string], ICritterDetailedResponse, unknown>;
  selectedAnimalFromSurveyCritterId: number | null;
  setSelectedAnimalFromSurveyCritterId: (selectedAnimalFromSurveyCritterId: number) => void;
};

export const AnimalPageContext = createContext<IAnimalPageContext | undefined>(undefined);

export const AnimalPageContextProvider = (props: PropsWithChildren<Record<never, any>>) => {
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [isDisabled, setIsDisabled] = useState<boolean>(false);

  const critterbaseApi = useCritterbaseApi();
  const biohubApi = useBiohubApi();

  const { surveyId, projectId } = useSurveyContext();

  const critterDataLoader = useDataLoader(critterbaseApi.critters.getDetailedCritter);
  const surveyCrittersDataLoader = useDataLoader(() => biohubApi.survey.getSurveyCritters(projectId, surveyId));

  if (!surveyCrittersDataLoader.data) {
    surveyCrittersDataLoader.load();
  }

  /**
   * Keeps track of which animal has been selected in the side panel of the animal page
   */
  const [selectedAnimal, setSelectedAnimal] = useState<ISurveyCritter | null>(null);

  /**
   * Used to set selectedAnimal from just surveyCritterId. Triggers a query for an
   * animal's Critterbase ID to update selectedAnimal.
   */
  const [selectedAnimalFromSurveyCritterId, setSelectedAnimalFromSurveyCritterId] = useState<number | null>(null);

  useEffect(() => {
    if (surveyCrittersDataLoader.data) {
      const critter = surveyCrittersDataLoader.data.find(
        (critter) => critter.survey_critter_id === selectedAnimalFromSurveyCritterId
      );

      if (critter?.critter_id && selectedAnimalFromSurveyCritterId) {
        animalPageContext.setSelectedAnimal({
          survey_critter_id: selectedAnimalFromSurveyCritterId,
          critterbase_critter_id: critter?.critter_id
        });
      }
    }
  }, [selectedAnimalFromSurveyCritterId, surveyCrittersDataLoader.data]);

  /**
   * Refreshes the current profile whenever the selected critter changes
   */
  useEffect(() => {
    if (selectedAnimal) {
      critterDataLoader.refresh(selectedAnimal.critterbase_critter_id);
    }
  }, [selectedAnimal]);

  const animalPageContext: IAnimalPageContext = useMemo(
    () => ({
      isLoading,
      setIsLoading,
      isDisabled,
      setIsDisabled,
      selectedAnimal,
      setSelectedAnimal,
      critterDataLoader,
      selectedAnimalFromSurveyCritterId,
      setSelectedAnimalFromSurveyCritterId
    }),
    [selectedAnimal, isLoading, isDisabled, critterDataLoader, selectedAnimalFromSurveyCritterId, setSelectedAnimalFromSurveyCritterId, setSelectedAnimal]
  );

  return <AnimalPageContext.Provider value={animalPageContext}>{props.children}</AnimalPageContext.Provider>;
};
