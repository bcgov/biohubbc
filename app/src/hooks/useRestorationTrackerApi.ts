import axios from 'axios';
import useAdminApi from './api/useAdminApi';
import useAxios from './api/useAxios';
import useCodesApi from './api/useCodesApi';
import useDraftApi from './api/useDraftApi';
import useExternalApi from './api/useExternalApi';
import useProjectApi, { usePublicProjectApi } from './api/useProjectApi';
import useSearchApi, { usePublicSearchApi } from './api/useSearchApi';
import useUserApi from './api/useUserApi';
import usePermitApi from './api/usePermitApi';
import { useContext } from 'react';
import { ConfigContext } from 'contexts/configContext';

/**
 * Returns a set of supported api methods.
 *
 * @return {*} object whose properties are supported api methods.
 */
export const useRestorationTrackerApi = () => {
  const config = useContext(ConfigContext);
  const apiAxios = useAxios(config?.API_HOST);

  const project = useProjectApi(apiAxios);

  const permit = usePermitApi(apiAxios);

  const search = useSearchApi(apiAxios);

  const codes = useCodesApi(apiAxios);

  const draft = useDraftApi(apiAxios);

  const user = useUserApi(apiAxios);

  const admin = useAdminApi(apiAxios);

  const external = useExternalApi(axios);

  const publicApis = {
    project: usePublicProjectApi(apiAxios),
    search: usePublicSearchApi(apiAxios)
  };

  return {
    project,
    permit,
    search,
    codes,
    draft,
    user,
    admin,
    external,
    public: publicApis
  };
};
