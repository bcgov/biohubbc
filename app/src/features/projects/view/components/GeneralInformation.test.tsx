import { cleanup, fireEvent, render, waitFor } from '@testing-library/react';
import { useRestorationTrackerApi } from 'hooks/useRestorationTrackerApi';
import { UPDATE_GET_ENTITIES } from 'interfaces/useProjectApi.interface';
import { getProjectForViewResponse } from 'test-helpers/project-helpers';
import React from 'react';
import { codes } from 'test-helpers/code-helpers';
import ProjectDetails from './GeneralInformation';
import { DialogContextProvider } from 'contexts/dialogContext';

jest.mock('../../../../hooks/useRestorationTrackerApi');
const mockuseRestorationTrackerApi = {
  project: {
    getProjectForUpdate: jest.fn<Promise<object>, []>(),
    updateProject: jest.fn()
  }
};

const mockRestorationTrackerApi = ((useRestorationTrackerApi as unknown) as jest.Mock<
  typeof mockuseRestorationTrackerApi
>).mockReturnValue(mockuseRestorationTrackerApi);

const mockRefresh = jest.fn();

const renderContainer = () => {
  return render(
    <DialogContextProvider>
      <ProjectDetails projectForViewData={getProjectForViewResponse} codes={codes} refresh={mockRefresh} />
    </DialogContextProvider>
  );
};

describe('ProjectDetails', () => {
  beforeEach(() => {
    // clear mocks before each test
    mockRestorationTrackerApi().project.getProjectForUpdate.mockClear();
    mockRestorationTrackerApi().project.updateProject.mockClear();
  });

  afterEach(() => {
    cleanup();
  });

  it('renders correctly with no end date (only start date)', () => {
    const { asFragment } = render(
      <ProjectDetails
        projectForViewData={{
          ...getProjectForViewResponse,
          project: { ...getProjectForViewResponse.project, end_date: (null as unknown) as string }
        }}
        codes={codes}
        refresh={mockRefresh}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with no activity data', () => {
    const { asFragment } = render(
      <ProjectDetails
        projectForViewData={{
          ...getProjectForViewResponse,
          project: { ...getProjectForViewResponse.project, project_activities: [] }
        }}
        codes={codes}
        refresh={mockRefresh}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with activity data', () => {
    const { asFragment } = renderContainer();

    expect(asFragment()).toMatchSnapshot();
  });

  it('editing the project details works in the dialog', async () => {
    mockRestorationTrackerApi().project.getProjectForUpdate.mockResolvedValue({
      project: {
        project_name: 'project name',
        project_type: 1,
        project_activities: [1, 2],
        start_date: '2020-04-20',
        end_date: '2020-05-20',
        revision_count: 2
      }
    });

    const { getByText, queryByText } = renderContainer();

    await waitFor(() => {
      expect(getByText('General Information')).toBeVisible();
    });

    fireEvent.click(getByText('Edit'));

    await waitFor(() => {
      expect(mockRestorationTrackerApi().project.getProjectForUpdate).toBeCalledWith(getProjectForViewResponse.id, [
        UPDATE_GET_ENTITIES.project
      ]);
    });

    await waitFor(() => {
      expect(getByText('Edit General Information')).toBeVisible();
    });

    fireEvent.click(getByText('Cancel'));

    await waitFor(() => {
      expect(queryByText('Edit General Information')).not.toBeInTheDocument();
    });

    fireEvent.click(getByText('Edit'));

    await waitFor(() => {
      expect(getByText('Edit General Information')).toBeVisible();
    });

    fireEvent.click(getByText('Save Changes'));

    await waitFor(() => {
      expect(mockRestorationTrackerApi().project.updateProject).toHaveBeenCalledTimes(1);
      expect(mockRestorationTrackerApi().project.updateProject).toBeCalledWith(getProjectForViewResponse.id, {
        project: {
          project_name: 'project name',
          project_type: 1,
          project_activities: [1, 2],
          start_date: '2020-04-20',
          end_date: '2020-05-20',
          revision_count: 2
        }
      });

      expect(mockRefresh).toBeCalledTimes(1);
    });
  });

  it('displays an error dialog when fetching the update data fails', async () => {
    mockRestorationTrackerApi().project.getProjectForUpdate.mockResolvedValue({
      project: undefined
    });

    const { getByText, queryByText } = renderContainer();

    await waitFor(() => {
      expect(getByText('General Information')).toBeVisible();
    });

    fireEvent.click(getByText('Edit'));

    await waitFor(() => {
      expect(getByText('Error Editing General Information')).toBeVisible();
    });

    fireEvent.click(getByText('Ok'));

    await waitFor(() => {
      expect(queryByText('Error Editing General Information')).not.toBeInTheDocument();
    });
  });

  it('shows error dialog with API error message when getting details data for update fails', async () => {
    mockRestorationTrackerApi().project.getProjectForUpdate = jest.fn(() =>
      Promise.reject(new Error('API Error is Here'))
    );

    const { getByText, queryByText, getAllByRole } = renderContainer();

    await waitFor(() => {
      expect(getByText('General Information')).toBeVisible();
    });

    fireEvent.click(getByText('Edit'));

    await waitFor(() => {
      expect(queryByText('API Error is Here')).toBeInTheDocument();
    });

    // Get the backdrop, then get the firstChild because this is where the event listener is attached
    //@ts-ignore
    fireEvent.click(getAllByRole('presentation')[0].firstChild);

    await waitFor(() => {
      expect(queryByText('API Error is Here')).toBeNull();
    });
  });

  it('shows error dialog with API error message when updating details data fails', async () => {
    mockRestorationTrackerApi().project.getProjectForUpdate.mockResolvedValue({
      project: {
        project_name: 'project name',
        project_type: 1,
        project_activities: [1, 2],
        start_date: '2020-04-20',
        end_date: '2020-05-20',
        revision_count: 2
      }
    });
    mockRestorationTrackerApi().project.updateProject = jest.fn(() => Promise.reject(new Error('API Error is Here')));

    const { getByText, queryByText } = renderContainer();

    await waitFor(() => {
      expect(getByText('General Information')).toBeVisible();
    });

    fireEvent.click(getByText('Edit'));

    await waitFor(() => {
      expect(mockRestorationTrackerApi().project.getProjectForUpdate).toBeCalledWith(getProjectForViewResponse.id, [
        UPDATE_GET_ENTITIES.project
      ]);
    });

    await waitFor(() => {
      expect(getByText('Edit General Information')).toBeVisible();
    });

    fireEvent.click(getByText('Save Changes'));

    await waitFor(() => {
      expect(queryByText('API Error is Here')).toBeInTheDocument();
    });

    fireEvent.click(getByText('Ok'));

    await waitFor(() => {
      expect(queryByText('API Error is Here')).toBeNull();
    });
  });
});
