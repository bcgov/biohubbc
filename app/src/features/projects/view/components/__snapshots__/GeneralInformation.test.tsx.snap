// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDetails renders correctly with activity data 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-5"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular"
      style="justify-content: space-between;"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit"
      >
        General Information
      </h3>
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <button
          aria-label="Edit General Information"
          class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
          data-testid="h3-button-toolbar-Edit"
          id="h3-button-toolbar-Edit"
          tabindex="0"
          title="Edit General Information"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                role="presentation"
                style="width: 1.3125rem; height: 1.3125rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                  style="fill: currentColor;"
                />
              </svg>
            </span>
            <strong>
              Edit
            </strong>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <dl>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Project Name
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            Test Project Name
          </dd>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Project Type
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            1
          </dd>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Timeline
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            Oct 10, 1998 - Feb 26, 2021
          </dd>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Activities
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            Activity code
          </dd>
        </div>
      </div>
    </dl>
  </div>
</DocumentFragment>
`;

exports[`ProjectDetails renders correctly with no activity data 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular"
      style="justify-content: space-between;"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit"
      >
        General Information
      </h3>
      <div
        class="MuiBox-root MuiBox-root-4"
      >
        <button
          aria-label="Edit General Information"
          class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
          data-testid="h3-button-toolbar-Edit"
          id="h3-button-toolbar-Edit"
          tabindex="0"
          title="Edit General Information"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                role="presentation"
                style="width: 1.3125rem; height: 1.3125rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                  style="fill: currentColor;"
                />
              </svg>
            </span>
            <strong>
              Edit
            </strong>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <dl>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Project Name
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            Test Project Name
          </dd>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Project Type
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            1
          </dd>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Timeline
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            Oct 10, 1998 - Feb 26, 2021
          </dd>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Activities
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            No Activities
          </dd>
        </div>
      </div>
    </dl>
  </div>
</DocumentFragment>
`;

exports[`ProjectDetails renders correctly with no end date (only start date) 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular"
      style="justify-content: space-between;"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit"
      >
        General Information
      </h3>
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <button
          aria-label="Edit General Information"
          class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
          data-testid="h3-button-toolbar-Edit"
          id="h3-button-toolbar-Edit"
          tabindex="0"
          title="Edit General Information"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                role="presentation"
                style="width: 1.3125rem; height: 1.3125rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                  style="fill: currentColor;"
                />
              </svg>
            </span>
            <strong>
              Edit
            </strong>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <dl>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Project Name
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            Test Project Name
          </dd>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Project Type
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            1
          </dd>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Timeline
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            <span>
              Start Date:
            </span>
             Oct 10, 1998
          </dd>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <dt
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Activities
          </dt>
          <dd
            class="MuiTypography-root MuiTypography-body1"
          >
            Activity code
          </dd>
        </div>
      </div>
    </dl>
  </div>
</DocumentFragment>
`;
